code/modules/client/client_procs.dm

```diff
@@ -1036,18 +1036,25 @@ GLOBAL_LIST_INIT(blacklisted_builds, list(
 				if("South")
 					movement_keys[key] = SOUTH
 				if("Say")
-					// I'm so sorry for anyone that has to read this. This is strung together with trial & error.
-					var/say = "\".winset \\\"command=\\\".start_thinking say\\\";tgui_modal.is-visible=true;\\\"\""
+					var/say = create_window_message(SAY_CHAN)
```

Comment:
theres no way this keeps this proc client side, right? Im pretty sure client verbs still query the server

Reply:
This should be just at keybinding assignment, the actual keystroke is using the string for .output (client side)
With this it's going key -> JS -> typing indicator
Previous was key -> indicator -> JS
I'll check in a minute

Reply:
and yes, this only runs on keybind init.

Reply:
If you can help it, set these up on the client (js), because if that say window stops working, people won't be able to OOC at all. Treat this as a fallback solution.